Applications/Spire/Source/TimeAndSales/TimeAndSalesTableView.cpp

```diff
@@ -0,0 +1,164 @@
+      set(BorderBottomSize(scale_height(1))).
+      set(BorderBottomColor(QColor(0xE0E0E0)));
+  }
+
+  auto apply_table_cell_style(StyleSheet& style) {
+    style.get(Any()).
+      set(border_size(0)).
+      set(horizontal_padding(scale_width(2))).
+      set(vertical_padding(scale_height(1.5)));
+  }
+
+  auto apply_table_cell_right_align_style(StyleSheet& style) {
+    style.get(Any()).set(TextAlign(Qt::AlignRight | Qt::AlignVCenter));
+  }
+
+  auto make_header_model() {
+    auto model = std::make_shared<ArrayListModel<TableHeaderItem::Model>>();
+    auto push = [&] (const QString& name, const QString& short_name) {
+      model->push({name, short_name, TableHeaderItem::Order::UNORDERED,
+        TableFilter::Filter::NONE});
+    };
+    push(QObject::tr("Time"), QObject::tr("Time"));
+    push(QObject::tr("Price"), QObject::tr("Px"));
+    push(QObject::tr("Size"), QObject::tr("Sz"));
+    push(QObject::tr("Market"), QObject::tr("Mkt"));
+    push(QObject::tr("Condition"), QObject::tr("Cond"));
+    push("", "");
+    return model;
+  }
+
+  auto make_header_item_properties() {
+    auto properties = std::vector<HeaderItemProperties>();
+    properties.emplace_back(false, Qt::AlignLeft, scale_width(48));
+    properties.emplace_back(true, Qt::AlignRight, scale_width(50));
+    properties.emplace_back(true, Qt::AlignRight, scale_width(40));
+    properties.emplace_back(true, Qt::AlignLeft, scale_width(38));
+    properties.emplace_back(false, Qt::AlignLeft, scale_width(34));
+    return properties;
+  }
+
+  auto make_time_cell(ptime time) {
+    auto time_text = to_text(time);
+    time_text = time_text.left(time_text.lastIndexOf('.'));
+    return make_label(time_text);
+  }
+
+  QWidget* table_view_builder(const std::shared_ptr<TableModel>& table, int row,
+      int column) {
+    auto column_id = static_cast<TimeAndSalesTableModel::Column>(column);
+    auto cell = [&] () -> QWidget* {
+      if(column_id == TimeAndSalesTableModel::Column::TIME) {
+        return make_time_cell(table->get<ptime>(row, column));
+      } else if(column_id == TimeAndSalesTableModel::Column::PRICE) {
+        auto money_cell = make_label(to_text(table->get<Money>(row, column)));
+        update_style(*money_cell, apply_table_cell_right_align_style);
+        return money_cell;
+      } else if(column_id == TimeAndSalesTableModel::Column::SIZE) {
+        auto quantity_cell = make_label(
+          to_text(table->get<Quantity>(row, column)).remove(QChar(',')));
+        update_style(*quantity_cell, apply_table_cell_right_align_style);
+        return quantity_cell;
+      } else if(column_id == TimeAndSalesTableModel::Column::MARKET) {
+        return make_label(
+          QString::fromStdString(table->get<std::string>(row, column)));
+      } else if(column_id == TimeAndSalesTableModel::Column::CONDITION) {
+        return make_label(
+          to_text(table->get<TimeAndSale::Condition>(row, column)));
+      }
+      return make_label("");
```

Comment:
What is this empty label for?

Reply:
It is used for the last empty column. It should use an empty `QWidget` instead of a label.